gvnitd2240/FileUploader
# Add Checkstyle Linter for Code Formatting and Enforce Style Checks on PRs

This PR introduces a Checkstyle linter to ensure code follows the required formatting and style guidelines. The linter will run automatically during each pull request, helping to maintain code quality and consistency across the project.

Changes:

	•	Added Maven plugins for Checkstyle and Spotless to enforce code style checks and formatting.
	•	Configured GitHub Actions to run style checks (spotless:check and checkstyle:check) on each pull request.
	•	Ensured that PRs cannot be merged unless the code passes the style checks.

Benefits:

	•	Ensures consistent code formatting and styling across the project.
	•	Automates the style-checking process, reducing manual reviews.
	•	Prevents PR merges if code style violations are present.

How to Test:

	•	Run mvn spotless:check checkstyle:check locally to verify code formatting.
	•	Raise a PR and confirm that GitHub Actions run style checks automatically.
